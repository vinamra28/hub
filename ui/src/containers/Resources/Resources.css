.hub-resource-card {
  background-color: white;
  height: 15em;
  width: 15em;
}
