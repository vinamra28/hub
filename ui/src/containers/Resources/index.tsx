import React from 'react';
import {
  Gallery,
  GalleryItem,
  Card,
  CardBody,
  Pagination,
  GridItem,
  Grid
} from '@patternfly/react-core';
import { Link } from 'react-router-dom';
import './Resources.css';

const Resources = () => {
  const [pageNumber, setPageNumber] = React.useState(1);
  const [perPage, setPerPage] = React.useState(20);

  const setPage = (event: React.MouseEvent | React.KeyboardEvent | MouseEvent, page: number) => {
    setPageNumber(page);
  };

  const perPageSelect = (
    event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
    perpage: number
  ) => {
    setPerPage(perpage);
  };
  return (
    <React.Fragment>
      <Grid>
        <GridItem>
          <Pagination
            itemCount={200}
            perPage={perPage}
            onSetPage={setPage}
            onPerPageSelect={perPageSelect}
            page={pageNumber}
            isCompact
          />

          <Gallery hasGutter>
            {Array.apply(0, Array(20)).map((x, i) => (
              <GalleryItem key={i}>
                <Link to="/tekton/task/name/version">
                  <Card isHoverable className="hub-resource-card">
                    <CardBody>Resources</CardBody>
                  </Card>
                </Link>
              </GalleryItem>
            ))}
          </Gallery>
        </GridItem>
      </Grid>
    </React.Fragment>
  );
};

export default Resources;
