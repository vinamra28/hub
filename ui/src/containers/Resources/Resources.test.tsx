import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Resources from '.';

it('should render the resources component', () => {
  const tree = renderer
    .create(
      <Router>
        <Resources />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
