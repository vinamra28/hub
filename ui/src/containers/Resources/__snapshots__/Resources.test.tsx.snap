// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the resources component 1`] = `
<div
  className="pf-l-grid"
>
  <div
    className="pf-l-grid__item"
  >
    <div
      className="pf-c-pagination pf-m-compact"
      data-ouia-component-id={0}
      data-ouia-component-type="PF4/Pagination"
      data-ouia-safe={true}
      id="pagination-options-menu-0"
    >
      <div
        className="pf-c-pagination__total-items"
      >
        <b>
          1
           - 
          20
        </b>
         
        of 
        <b>
          200
        </b>
         
        
      </div>
      <div
        className="pf-c-options-menu"
        data-ouia-component-id={1}
        data-ouia-component-type="PF4/PaginationOptionsMenu"
        data-ouia-safe={true}
      >
        <div
          className="pf-c-options-menu__toggle pf-m-plain pf-m-text"
        >
          <span
            className="pf-c-options-menu__toggle-text"
          >
            <b>
              1
               - 
              20
            </b>
             
            of 
            <b>
              200
            </b>
             
            
          </span>
          <button
            aria-expanded={false}
            aria-label="Items per page"
            className="  pf-c-options-menu__toggle-button"
            disabled={false}
            id="pagination-options-menu-toggle-0"
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <span
              className="pf-c-options-menu__toggle-button-icon"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  transform=""
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <nav
        aria-label="Pagination"
        className="pf-c-pagination__nav"
      >
        <div
          className="pf-c-pagination__nav-control"
        >
          <button
            aria-disabled={true}
            aria-label="Go to previous page"
            className="pf-c-button pf-m-plain pf-m-disabled"
            data-action="previous"
            data-ouia-component-id={2}
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={true}
            onClick={[Function]}
            tabIndex={null}
            type="button"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                transform=""
              />
            </svg>
          </button>
        </div>
        <div
          className="pf-c-pagination__nav-control"
        >
          <button
            aria-disabled={false}
            aria-label="Go to next page"
            className="pf-c-button pf-m-plain"
            data-action="next"
            data-ouia-component-id={3}
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                transform=""
              />
            </svg>
          </button>
        </div>
      </nav>
    </div>
    <div
      className="pf-l-gallery pf-m-gutter"
    >
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={4}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={5}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={6}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={7}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={8}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={9}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={10}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={11}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={12}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={13}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={14}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={15}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={16}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={17}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={18}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={19}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={20}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={21}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={22}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
      <div>
        <a
          href="/tekton/task/name/version"
          onClick={[Function]}
        >
          <article
            className="pf-c-card pf-m-hoverable hub-resource-card"
            data-ouia-component-id={23}
            data-ouia-component-type="PF4/Card"
            data-ouia-safe={true}
          >
            <div
              className="pf-c-card__body"
            >
              Resources
            </div>
          </article>
        </a>
      </div>
    </div>
  </div>
</div>
`;
